openhands/runtime/utils/request.py

```
@@ -21,15 +21,15 @@ def __str__(self) -> str:
         return s
 
 
-def is_rate_limit_error(exception):
+def is_retryable_error(exception):
     return (
         isinstance(exception, requests.HTTPError)
         and exception.response.status_code == 429
-    )
+    ) or isinstance(exception, requests.ConnectionError)
```

Can we not do this? Why is it needed?

429 makes sense, connection error should be a hard fail

---

Even a connection error could be transient, no?

---

From my experience -- connection error can actually be transcient, it happens once maybe per 1000 requests, but when it happens the whole thing failed :(

---

especially in eval -- when it happens, it will terminate the ENTIRE session that have been running and everything have to start from scratch again, which may end-up costing a lot more $$

---

ok.. it seems retry to connection error caused issues for UI itself (eg error creating agent controller). Reverted and it work, I guess I'll probably add it to remote_runtime exec_action, instead of in this util.

---

With a 429 error, you _expect_ it to recover after some period of time, so it makes sense to retry

With connection and other errors, maybe it'll recover, maybe it won't. Generally best to fail hard, and handle retries as far back in the stack as you can. Otherwise the client is forced to sit there waiting for a connection that will never materialize, which can be super frustrating